tests/Functional/Platform/RenameColumnTest.php

```diff
@@ -33,6 +35,35 @@ public function testColumnPositionRetainedAfterRenaming(string $columnName, stri
 
         $table = $sm->introspectTable('test_rename');
         self::assertSame([strtolower($newColumnName), 'c2'], array_keys($table->getColumns()));
+        self::assertCount(1, $diff->renamedColumns);
+    }
+
+    /** @dataProvider columnNameProvider */
+    public function testColumnPositionRetainedAfterExplicitRenaming(string $columnName, string $newColumnName): void
+    {
+        $table = new Table('test_rename');
+        $table->addColumn($columnName, Types::INTEGER, ['length' => 16]);
+        $table->addColumn('c2', Types::INTEGER);
+
+        $this->dropAndCreateTable($table);
+
+        // Force a different type to make sure it's not being caught implicitly
+        $table->renameColumn($columnName, $newColumnName)->setType(Type::getType(Types::BIGINT))->setLength(32);
```

Comment:
It's worth noting that some drivers will fail to change some types to incompatible types, it doesn't have anything to do with the column rename and just changing the type of a column in general, the biggest offender being DB2 is unable to transform a `string`  type to a `text` type, it's impossible to know this in advance and unlikely we could cover all edge cases, so I'm leaving it to the driver to throw an error when the query fails in that case (as was previously the case)

The solution is to add a temp column, move the data there, drop the old column, recreate it with the new type and move the data again but it's also not something that will always work because some types are just not compatible, that's a loop of hoops and potential breaking points that will leave the database in an invalid state for little benefit and I would not attempt to automate it

It's not a blocker for this PR and just a possible future improvement to the offending drivers if the need really arises